lib/modules/manager/swift/extract.spec.ts

```diff
@@ -0,0 +1,247 @@
+import { extractPackageFile } from './extract';
+import { Fixtures } from '~test/fixtures';
+
+const samplePackageSwift = Fixtures.get('SamplePackage.swift');
+
+describe('modules/manager/swift/extract', () => {
+  describe('extractPackageFile()', () => {
+    it('returns null for empty content', () => {
+      expect(extractPackageFile('')).toBeNull();
+    });
+
+    it('returns null for content without dependencies', () => {
+      const content = `
+        let package = Package(
+          name: "MyPackage",
+          products: [
+            .library(
+              name: "MyLibrary",
+              targets: ["MyLibrary"]
+            )
+          ],
+          targets: [
+            .target(
+              name: "MyLibrary"
+            )
+          ]
+        )
+      `;
+      expect(extractPackageFile(content)).toBeNull();
+    });
+
+    it('extracts GitHub dependencies with github-tags datasource', () => {
+      const content = `
+        let package = Package(
+          name: "MyPackage",
+          dependencies: [
+            .package(url: "https://github.com/example/repo", from: "1.0.0")
+          ]
+        )
+      `;
+      const result = extractPackageFile(content);
+      expect(result).toMatchObject({
+        deps: [
+          {
+            datasource: 'github-tags',
+            depName: 'example/repo',
+            packageName: 'https://github.com/example/repo',
```

Comment:
should be dropped, datasource defaults to github.com

Reply:
I misunderstood your comments, I'll change to remove `packageName` field.